### PR TITLE
[3.13, dynamo] support TO_BOOL

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #130751
* #130569
* #130568
* #130567
* #130566
* __->__ #130565
* #130564
* #130461
* #130460
* #130459

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -2262,6 +2262,14 @@ def STORE_FAST_STORE_FAST(self, inst):
     def CALL_KW(self, inst):
         self._call(inst, call_kw=True)
 
+    @break_graph_if_unsupported(push=1)
+    def TO_BOOL(self, inst):
+        value = self.pop()
+        if value.is_python_constant():
+            self.push(ConstantVariable(bool(value.as_python_constant())))
+        else:
+            self.push(BuiltinVariable(bool).call_function(self, [value], {}))
```
Comment (jansel) [inline torch/_dynamo/symbolic_convert.py:2271]: Shouldn't this be able to handle the constant case as well?

**Review by jansel — CHANGES_REQUESTED** [review]

**Comment by williamwen42** [comment]
Changed approach: we don't actually need to do anything for `TO_BOOL`, based on how CPython codegens this instruction.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**